common/src/core/java/xyz/wagyourtail/jsmacros/core/library/impl/FJsMacros.java

```diff
@@ -542,6 +602,14 @@ public Thread getCreator() {
                     return null;
                 }
 
+                @Override
+                public void off() {
+                    lock.release();
+                    lock2.release();
+                    done[0] = true;
+                    Core.getInstance().eventRegistry.removeListener(event, this);
```

Comment:
can we get this to throw an interrupted exception on the waitForEvent's thread instead somehow?

Reply:
I'm not that familiar with threads. I tried `th.interrupt()` and removed the releases and it seems to work as expected. The waitForEvent throws an exception and if catched the script continues running, just like any other async functions.
```java
Core.getInstance().eventRegistry.removeListener(event, this);
th.interrupt();
```
The test script
```js
JavaWrapper.methodToJavaAsync(() => {
	Client.waitTick(20);
	Chat.log("Running");
}).run();

JavaWrapper.methodToJavaAsync(() => {
	Client.waitTick(5);
	JsMacros.disableScriptListeners();
}).run();

try {
	JsMacros.waitForEvent("Sound", JavaWrapper.methodToJava(e => {
		Chat.log("Event triggered");
		return true;
	}));
} catch (e) {
	Chat.log(e);
}

Chat.log("Finished");
```
It will log:
java.lang.InterruptedException
Finished
Running

Reply:
Maybe add this as a unit test

Reply:
also, does this throw the runtime exception with "error thrown in filter"? might want to make it a different catch block for interrupted exception so we can pass it as-is

Reply:
Doing something like this will throw an error in the filter, but still log `Event triggered` before the Exception is thrown
```java
	JsMacros.waitForEvent("Sound", JavaWrapper.methodToJava(e => {
		JsMacros.disableScriptListeners();
		Chat.log("Event triggered");
		return true;
	}));
```
With the previous code it won't throw the error from inside the filter.

Reply:
no. I mean that interrupting elsewhere would throw at https://github.com/JsMacros/JsMacros/blob/aa6c3ff006cde39a8f842b267821afedbe7d1b59/common/src/core/java/xyz/wagyourtail/jsmacros/core/library/impl/FJsMacros.java#L639 and I think it should have a seperate catch statement / throw the interrupted exception out of the function

Reply:
o I'm wrong